File: Data Structures & Algorithms/two-integer-sum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number[]}
     */
    twoSum(nums, target) {
        const len = nums.length;

        for(let i=0; i<len; i++){
            for(let j=i+1; j<len; j++){
                if(nums[i]+nums[j] === target) return [i,j];
            }
        }
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // left pointer .. 0, right pointer .. n-1
        // if sum(l+r)
        // < target; move left pointer 
        // > targer; move right pointer
        int len = numbers.size();
        int l = 0, r = len - 1;


        while(l < r){
            int sum = numbers[l] + numbers[r];
            if(sum < target) { l++; }
            else if(sum > target) { r--; }
            else { break; } 
        }

        vector<int> res = {l+1, r+1};

        return res;
    }
};
